### PR TITLE
Dark Theme Visibility Fixes

### PR DESCRIPTION
**Pull request by ageofadz**
There had been some visibility issues with the interface when in dark theme.

- Changed the color of the vertical scroll bar in dark mode to allow it to be more visible against the other colors.

- Changed the color of the accent behind the number selected which appears in the group tree when an item in a group is selected to allow for the number inside to be more readable.

fixes [#5522](https://github.com/JabRef/jabref/issues/5522)

Then:

Now:
Screen Shot 2019-11-19 at 5 44 54 PM
----

- [x ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [x] Screenshots added in PR description (for bigger UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

**Comment by ageofadz** [comment]
Sorry if I did anything wrong, this is my first pull request on a public repo.

Also happy birthday if you're reading this dad!

File: scripts/generate-authors.sh [hunk 1]
```diff
@@ -52,6 +52,7 @@ cd "$(dirname "$(readlink -f "$BASH_SOURCE")")/.."
 	Samin Muhammad Ridwanul Karim
 	Stefan Robert
 	Bernhard Tempel
+	Samuel Robertson
```
Comment (koppor) [inline scripts/generate-authors.sh]: No need to add yourself. You'll be added correctly by `generate-authors.sh` based on the data in your commit, In this place in `generate-authors.sh`, the authors contributing in our [subversion](https://subversion.apache.org/) times on sourceforge are tracked. With git, we don't need that system any more.

**Comment by koppor** [comment]
@ageofadz Thank you for being brave and submitting a pull request. This is how open source should work 🎉 

Regarding the red cross. It's because our test coverage checker sometimes fails due to time outs. You can ignore this error. (Refs https://github.com/JabRef/jabref/pull/5618).

**Comment by Siedlerchr** [comment]
Thank you very much for your contribution! 🥇  Looks good to me! Would you mind fixing the preview (ciation styles) panel for OpenOffice/LibreOffice as well? This is the same one as in the entry editor, so it should be easy to adapt.

**Comment by AEgit** [comment]
Just for clarification: @Siedlerchr is referring to the issue mentioned here:
https://github.com/JabRef/jabref/issues/5522#issuecomment-547054970

File: src/main/java/org/jabref/gui/Dark.css [hunk 2]
```diff
@@ -64,3 +64,16 @@
     -fx-background-color: -fx-light-text-color, -fx-control-inner-background;
 }
 
+.scroll-bar:horizontal .track,
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/Dark.css]: Similar code is already in the `Base.css` file:
https://github.com/JabRef/jabref/blob/5376dcfb648da7ad43606694da841b569b7bfb26/src/main/java/org/jabref/gui/Base.css#L704-L717

Thus, it should be sufficient if you overwrite the following color specifications:
https://github.com/JabRef/jabref/blob/5376dcfb648da7ad43606694da841b569b7bfb26/src/main/java/org/jabref/gui/Base.css#L89-L91
here in the `root` part of `Dark.css`.

File: src/main/java/org/jabref/gui/groups/GroupTree.css [hunk 3]
```diff
@@ -30,13 +30,12 @@
 }
 
 .numberColumn > .hits:any-selected {
-    -fx-background-color: derive(-jr-green, 70%);
+    -fx-background-color: derive(-jr-gray-3, 70%);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/groups/GroupTree.css]: With these changes, the number column is now also displayed in gray in the light theme. This is probably also not that readable.

I would propose to add two colors (`-jr-groups-any-hits` and `-jr-groups-all-hits`), which are used here. Then the dark theme can overwrite them.

**Comment by ageofadz** [comment]
> Thank you very much for your contribution! 🥇 Looks good to me! Would you mind fixing the preview (ciation styles) panel for OpenOffice/LibreOffice as well? This is the same one as in the entry editor, so it should be easy to adapt.


Thanks so much for the feedback everyone!
Started working on a few of the changes. As for the changes to the OpenOffice style selection, there's a method in /gui/preview/PreviewViewer.java called setTheme() that can apply a theme to previewArticle and previewBook in /gui/openoffice/StyleSelectDialogView.java, with a note about [an open issue on openJDK](https://github.com/openjdk/jfx/pull/22). I was able to add `previewArticle.setTheme("DARK_CSS");` to get the text in the OpenOffice style selection to fit the dark mode, but because it only takes strings, I haven't been able to get a string value for the current theme applied so that the style selection's appearance matches the theme selected.
Here's the method's code:

`    public void setTheme(String theme) {`
`        if (theme.equals(ThemeLoader.DARK_CSS)) {`
`            // We need to load the css file manually, due to a bug in the jdk`
`            // TODO: Remove this workaround as soon as https://github.com/openjdk/jfx/pull/22 is merged`
`            URL url = JabRefFrame.class.getResource(ThemeLoader.DARK_CSS);`
`            String dataUrl = "data:text/css;charset=utf-8;base64," +`
``                    `Base64.getEncoder().encodeToString(StringUtil.getResourceFileAsString(url).getBytes());`
``
`            previewView.getEngine().setUserStyleSheetLocation(dataUrl);`
`        }`
`    }`

**Comment by tobiasdiez** [comment]
What do you mean with the following?
> but because it only takes strings, I haven't been able to get a string value for the current theme applied so that the style selection's appearance matches the theme selected.

The method allows you to load the `Dark.css`, in which you then can add/change the necessary style information.

**Comment by ageofadz** [comment]
> What do you mean with the following?
> 
> > but because it only takes strings, I haven't been able to get a string value for the current theme applied so that the style selection's appearance matches the theme selected.
> 
> The method allows you to load the `Dark.css`, in which you then can add/change the necessary style information.

Because it checks if "Dark.css" is passed in as an argument and switches to dark theme if it is, it needs a string that says which theme is active for it to change according to what theme the user selected.
At the time I couldn't find one, but I found the hashmap in JabrRefPreferences and passed that into it now with `JabRefPreferences prefs = JabRefPreferences.getInstance();` and `previewArticle.setTheme(prefs.get(JabRefPreferences.FX_THEME));`, which appear to work!

Testing right now, will update my pull requests later today with that addition and your earlier suggestions.

**Comment by Siedlerchr** [comment]
>JabRefPreferences prefs = JabRefPreferences.getInstance(); and previewArticle.setTheme(prefs.get(JabRefPreferences.FX_THEME));, which appear to work!

In the Style Select DialogView you have access to the preferences via the Preferences Service which is a wrapper (for dependency injection via the FXML loader) for the JabRefPreferences.
We try to pass dependencies always as constructor arguments (Inversion of Control principle) to avoid direct dependency calls.
So the idea would be to add a method e.g. getActiveTheme to the PreferencesService which returns the theme for the preferences. Then you can call setTheme on the PreviewViewer.

**Comment by ageofadz** [comment]
Hey, many thanks once again for everyone's feedback!
The [newest commit](https://github.com/ageofadz/jabref/commit/aa06737230927cd0c9cd32a9860d26e20c367797) has updates to dark.css based on @tobiasdiaz's suggestion to overwrite scroll bar and group hit colors, in addition to @Siedlerchr's suggestion to update the StyleSelectDialogView using the Preferences Services wrapper injection as opposed to a hard dependency to JabRefPreferences.

Attached is a screenshot displaying the changes to scrollbar and group selected indicator (visually unchanged since first commit) and the OO style selection dialog view.

Screen Shot 2019-11-23 at 12 08 20 PM


[]  Tests created for changes
[x]  Manually tested changed features in running JabRef
[x]  Screenshots added in PR description (for bigger UI changes)
[x]  Checked documentation: Is the information available and up to date? If not: Issue created at https://github.com/JabRef/user-documentation/issues.

**Comment by tobiasdiez** [comment]
This looks good! Thanks for the follow-up.

I think you latest commit reverted the changes in the previous commit https://github.com/ageofadz/jabref/commit/b812b62b6b854795fc9c012aa458959ed581bcfe. You can use git cherry-pick to integrate these changes as well.

**Comment by ageofadz** [comment]
> This looks good! Thanks for the follow-up.
> 
> I think you latest commit reverted the changes in the previous commit [ageofadz@b812b62](https://github.com/ageofadz/jabref/commit/b812b62b6b854795fc9c012aa458959ed581bcfe). You can use git cherry-pick to integrate these changes as well.

To fix that, should I use `git cherry-pick ^aa06737230927cd0c9cd32a9860d26e20c367797..b812b62b6b854795fc9c012aa458959ed581bcfe`?
The commit from two hours ago should have just added getTheme() to preferencesServices and jabRefPreferences and updated the StyleSelectDialogView to remove directly calling jabRefPreferences to refer to the preferencesServices getTheme().

**Comment by tobiasdiez** [comment]
Mr. @koppor git wizard, can you please assist with the git issue.

**Comment by koppor** [comment]
If it is only the commit b812b62b6b854795fc9c012aa458959ed581bcfe, `git cherry-pick b812b62b6b854795fc9c012aa458959ed581bcfe` is enough (no need for commit ranges). Possibly, conflicts have to be resolved. Just resolve it with the IDE and then commit the result. I use `git gui` and click on the envelope buttons at the upper left.

**Comment by ageofadz** [comment]
Newest commit cherry-picks b812b62 and resolved conflicts in CHANGELOG.md and StyleSelectDialogView.java.
Thank you git wizard @koppor!

**Comment by koppor** [comment]
I think a `git push` is missing? Maybe you cherry-picked on the wrong branch? You can always check the current overall branch status with `gitk --all&`. Then, it displays where which branch currently points to. (Maybe the articleh https://lostechies.com/joshuaflanagan/2010/09/03/use-gitk-to-understand-git/ helps a bit?)

**Comment by ageofadz** [comment]
The latest commit is, 4411dceb783f1d16ceffae10e77b1c52fb2cf390. `gitk --all&`, shows it as HEAD with 508c2bedd0c70991fb2506f0cd6351f40026ca91 as parent, `git status` says I'm up to date.

Unsure why it isn't appearing in the PR, but appears on [my fork](https://github.com/ageofadz/jabref/commits/master)

**Comment by tobiasdiez** [comment]
You pushed it to your master branch but not to the base branch of this PR.

**Comment by tobiasdiez** [comment]
This looks good now. Thanks again and looking forward to your next PR 😄 !

**Comment by ageofadz** [comment]
Couldn’t wish for a better first PR. Learned a lot and had a blast, I’m stoked to do many more.
Thank you all very much for being so helpful and patient commit after commit 😂

**Pull request merged**